### PR TITLE
feat: match collection view to spec

### PR DESCRIPTION
**Pull request by zchsh**
## ✅ "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][asana] 🎟️

## 🗒️ What

This PR finishes the UI implementation of `views/collection-view`. With recent work completed in #363 , the only remaining UI change needed is to remove the `Tutorials` heading and the sidecar to match design specs.

## 🤷 Why

To match the `collection-view` implementation to design specs.

## 🛠️ How

- Makes a change in `SidebarSidecarLayout` to allow `sidecarSlot` to be falsy
- Passes `null` to `sidecarSlot` in order to not render a sidecar content
- Removes `heading` generation for `collection-view` as it is now unused
- Makes related TypeScript updates to removal of `headings` from `collection-view` props

## 📸 Design Screenshots

<details>
<summary>Implementation before this PR</summary>
before
</details>

<details>
<summary>Design spec</summary>
design-spec
</details>

<details>
<summary>Implementation after this PR</summary>
after
</details>

## 🧪 Testing

- [ ] Visit a collection view, such as [`/vault/tutorials/encryption-as-a-service`][preview]
- [ ] Ensure the `Tutorials` heading is no longer rendered, and the sidecar area is empty

## 💭 Anything else?

Not at the moment!

[preview]: https://dev-portal-git-zsfinish-collection-view-hashicorp.vercel.app/vault/tutorials/encryption-as-a-service
[asana]: https://app.asana.com/0/1202001387788447/1202168098250852/f

### CODE REVIEW

File: src/layouts/sidebar-sidecar/index.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const SidebarSidecarLayoutContent = ({
   }
 
   const SidecarContent = (): ReactElement => {
-    if (sidecarSlot) {
+    if (typeof sidecarSlot !== 'undefined') {
```
Comment (zchsh) [inline src/layouts/sidebar-sidecar/index.tsx:60]: I think we want to pass `null` to `sidecarSlot` to keep it empty, so updated this check to be a bit more precise.

File: src/views/collection-view/index.tsx [hunk 2]
```diff
@@ -28,12 +28,11 @@ function CollectionView({
       }
     >
       <CollectionMeta
-        heading={{ text: name, id: layoutProps.headings[0].slug }}
+        heading={{ text: name, id: 'overview' }}
```
Comment (zchsh) [inline src/views/collection-view/index.tsx:31]: Hard-coded this at this level, as `slug` in the underlying `Heading` component is a required prop.

File: src/views/collection-view/index.tsx [hunk 3]
```diff
@@ -28,12 +28,11 @@ function CollectionView({
...
         description={description}
         cta={{ href: getTutorialSlug(tutorials[0].slug, slug) }}
         numTutorials={tutorials.length}
       />
-      <h2 id={layoutProps.headings[1].slug}>Tutorials</h2>
```
Comment (zchsh) [inline src/views/collection-view/index.tsx:36]: Have removed this heading entirely, to match the design spec.

File: src/views/collection-view/server.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ export interface CollectionPageProps {
 
 export type CollectionLayout = Pick<
   SidebarSidecarLayoutProps,
-  'headings' | 'breadcrumbLinks'
```
Comment (zchsh) [inline src/views/collection-view/server.ts:29]: The `CollectionLayout` no longer uses `headings`, so I've dropped it from the type.

File: src/views/product-tutorials-view/server.ts [hunk 5]
```diff
@@ -25,7 +26,10 @@ export interface ProductTutorialsPageProps {
   product: ProductTutorialsPageProduct
 }
 
-type ProductTutorialsLayout = CollectionLayout
+type ProductTutorialsLayout = Pick<
```
Comment (zchsh) [inline src/views/product-tutorials-view/server.ts:29]: `ProductTutorialsLayout` previously mirrored `CollectionLayout`, but now it does not, as the former needs `headings` while the latter has no use for them.

**Comment by zchsh** [comment]
@kendallstrautman This handles the TODO mentioned in [this thread on #363](https://github.com/hashicorp/dev-portal/pull/363#discussion_r856669604).

Let me know what you think 🙇‍♂️

---

Lower priority note on `helper` & `util` functions shared across multiple components:

As in #363 with `getReadableTime`, I considered trying to move these out of specific `component` folders, as many of them are shared... but wasn't sure where we should put them... maybe `lib`? `lib/learn-client`? or maybe not in `learn-client` since they aren't really "the client"? 🤔 In the end I figured since there are quite a few we'd potentially want to move around (_if_ it makes sense), I'd wait to get your take on what we should do, and _if_ it feels like re-organization is worth it, then we could establish a shared folder and move a bunch of stuff around in a separate PR.

File: src/views/collection-view/index.tsx (hunk 2 above)
Comment (kendallstrautman) [reply src/views/collection-view/index.tsx:31]: You could also use `id` or `slug` from the `collection` prop

Comment (zchsh) [reply src/views/collection-view/index.tsx:31]: ah true, that might make it more clear that this now has nothing to do with sidecar - will update! 👍

**Comment by zchsh** [comment]
@kendallstrautman Made a task for myself for the `lib/tutorials` change - 🎟️ [[chore] make lib/tutorials for shared use tutorial helpers](https://app.asana.com/0/1202097197789424/1202180048371484/f). Going to merg this in, & follow up with that later 👍

**Pull request merged**